multicluster/controllers/multicluster/member/clusterset_controller.go

```diff
@@ -130,20 +154,57 @@ func (r *MemberClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			err = r.Update(context.TODO(), clusterSet)
 			if err != nil {
 				klog.ErrorS(err, "Failed to update ClusterSet's ClusterID", "clusterset", req.NamespacedName)
-				return ctrl.Result{}, err
+				return err
 			}
 		}
+
+		r.clusterSetConfig = clusterSet.DeepCopy()
```

Comment:
From what I saw seems only `clusterSet.Generation` is really useful to be saved. But then I got another question - probably we should save generation before `if !clusterSetCreated && !leaderChanged { return nil }`?

Reply:
Yeah, moved it.